src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java

```diff
@@ -98,18 +98,14 @@ public Path run() {
             URIList URIList = module.getUriList();
 
             enforceFileLimit(URIList, limit);
-            downloadFiles(URIList, tmpDir);
+            List<Path> downloadedFiles = downloadFiles(URIList, tmpDir);
             throwIfCancelled();
-            File tempDirFile = new File(tmpDir.toString());
-            File[] files = tempDirFile.listFiles();
-            if (files != null && files.length > 0) {
-                module.loadFileMetadata(files[0]);
-            }
-            applySubsetMultiThread(tmpDir, module, goGoDuckConfig.getThreadCount());
+            module.loadFileMetadata(downloadedFiles.get(0).toFile());
```

Comment:
Do you think it is a good idea to include the condition to check if at least one file is downloaded?

```
            if (files != null && files.length > 0) {
                 module.loadFileMetadata(files[0].toFile());
             }
```